ea-engine/ea.py

```diff
@@ -0,0 +1,82 @@
+from functools import partial
+import math
+import random
+from deap import algorithms, base, creator, tools
+import numpy as nn
+from operator import itemgetter
+
+class EA: 
+
+    def __init__(self, num_drivers, num_orders, data) -> None:  
+        creator.create("FitnessMin", base.Fitness, weights=(-1.0,))
+        creator.create("Individual", list, fitness=creator.FitnessMin)
+        self.__toolbox = base.Toolbox()
+        self.__toolbox.register("indices", random.uniform, 0, num_drivers)
+        self.__toolbox.register("individual", tools.initRepeat,
+                 creator.Individual, self.__toolbox.indices, n=num_orders)
+        self.__toolbox.register("population", tools.initRepeat, list, self.__toolbox.individual)
+        self.__toolbox.register("evaluate", self.__evaluation)
+        self.data = data
+
+    def set_crossover(self, crossover_type):
+        self.__toolbox.register("mate", crossover_type)
+        return self
+
+    def set_mutate(self, mutate_type, **kwargs):
+        self.__toolbox.register("mutate", mutate_type, **kwargs)
+        return self
+
+    def set_selection(self, selection_type, **kwargs):
+        self.__toolbox.register("select", selection_type, **kwargs)
+        return self
+
+    def create_population(self, num_individuals):
+        self.__pop = self.__toolbox.population(n=num_individuals)
+        return self
```

Comment:
Why this is not a part of `eval_mode`?
Maybe we should change this to `set_population` as all other parameters, and initial the population in the eval mode?